### ISSUE TITLE
Enable to get the possibility to save more then one picture

### ISSUE THREAD

**Issue by Githopp192** [nextcloud/cookbook#443]
**Is your feature request related to a problem? Please describe.**

no problem.

**Describe the solution you'd like**

Most of my recipies .. i'm taking some picture from my cookbooks & then i will put this picture into the 
NC App. This works fine. Thx 4 that !

On the other hand - In some cases, there were some cookbooks, where recipes are written over several pages, so i'd need to 
get more than one picture to include into the NC App - this would be a nice feature .. i'm thinking .. hm .. to save al least five pictures would be nice.

**Describe alternatives you've considered**
none :-)

**Comment by seyfeb** [comment]
Hi @Githopp192, would you mind elaborating on 

>  In some cases, there were some cookbooks, where recipes are written over several pages, so i'd need to
get more than one picture to include into the NC App

\- are you talking about printed cookbooks?

Is your goal to illustrate certain recipe steps/tools/ingredients or do you suggest to have general photos next to the main photo?

**Comment by Githopp192** [comment]
Hi @seyfeb  

yes .. printed cookbooks.

i think .. best explained is .. i put one example of my last recipes i put into the cookbook app :-)

There are three pages ... what i did as workaround .. i used an iPhone App (PicCollage --> https://apps.apple.com/ch/app/piccollage-grid-story-maker/id448639966) to have a workaround to get more than one picture.

![full](https://user-images.githubusercontent.com/32612067/102700433-8d91f500-424d-11eb-9985-44e1d27f6bfc.jpg)

**Comment by seyfeb** [comment]
In my opinion there is a requirement to have a paginated recipe view in the NC apps webview. In don’t see the point of creating a book-like appearance in the sense that you should have to change pages to be able to see the whole recipe.

As for having multiple images in general, there is a discussion on adding images to recipe steps in #47. For storing the recipe data, we follow the schema.org Recipe standard. I might be mistaken, but if I remember correctly, the schema.org format allows only for a single main image.

**Comment by christianlupus** [comment]
Also related: #324.

Regarding the storage: In general, the schema.org standard allows for more than one image. (FYI: The entries can always be repeated.) However, I see more of a UI problem here. How/where should the image be located (except for an image carousel on the top)? Keep in mind that the app is comparably dumb and designing a good-looking book is a complete job on its own (called lecturer any typographer).

So a few questions back to you to sort out the desired effect a bit:

1. How should the images be located/oriented?
2. How do you consider the Web-UI to be done? Maybe draw a sketch of a few screens (handwritten on paper and later photographed is perfectly fine)?

When we are printing from the browser, we are not so nice in the output due to headers/footers added by the browser. Better would maybe be a program tailored towards book-making. I am thinking of LaTeX and Scribus. However, this is out of scope especially as this would have to be installed on the server as a dependency which is obviously not the case.

**Comment by mottone** [comment]
> Is your goal to illustrate certain recipe steps/tools/ingredients or do you suggest to have general photos next to the main photo?

IMHO, it would be useful to illustrate steps/tools/ingredients just to make better explanation of the recipe.

**Comment by christianlupus** [comment]
> IMHO, it would be useful to illustrate steps/tools/ingredients just to make better explanation of the recipe.

I think there is a closer relation with #324, right?

> For my needs it would suffice to have the typical right/left arrows in the photo view to flip between images. Just being able to save the pictures somewhere would be great.

Regarding the storage we have already #340.
That would mean to have such a carousel as I suspected earlier.

> And while we are a it, I would also love to attach arbitrary files, not just images. But that will be a separate issue.

For that please open a new issue to avoid burying this in a stack of comments.

**Comment by mottone** [comment]
> > IMHO, it would be useful to illustrate steps/tools/ingredients just to make better explanation of the recipe.
> 
> I think there is a closer relation with #324, right?
> 
Yes, using Markdown would be a nice method

**Comment by ventilator** [comment]
Any progress if its possible to add more than one image?

**Comment by christianlupus** [comment]
This needs some preparations in the backend to reduce the total work involved to add new features.

**Comment by gsantner** [comment]
Is this somewhat planned to be added anytime soon? Being limited to one picture is quite restricting, also the schema-org format supports multiple media at one go. No video support either unfortunately.

Usually you have a recipe from _somewehere_ and want to add pictures of steps, variations or how it looked by your own cooking 😄 .

**Comment by christianlupus** [comment]
As already mentioned, this is not directly implementable. Some preparations need to be done that involve quite some workin the background. Sorry for that.

In general, we have multiple issues related to multi-media support:

- This Issue is related to have more than one recipe image.
- #206 is about adding per-recipe videos
- #47 is about images per step.

All three have in common that a storage infrastructure needs to be created that is long-lasting. That is, that the app can find/identify the images in an efficient way and that will not require rework in half a year.

**Comment by gsantner** [comment]
Media data could also be stored as media BLOB or base64 encoded in the Nextcloud database. That makes also the thumbnail images gone that currently show up in the filemanager (not sure if this information should be exposed as normal files via filemanager). Other Nextcloud apps like Music Players also have things like album cover.jpg's and they also don't save images to the normal files. That also makes sharing individual recipes with other nextcloud users possible I guess (user id to recipe mapping).

![Screenshot_20240111-204115](https://github.com/nextcloud/cookbook/assets/6735650/3f598603-1c9c-44c0-bb9b-212ade5c7a30)

![Screenshot_20240111-204101](https://github.com/nextcloud/cookbook/assets/6735650/7c7c74ee-4234-4a58-9904-fed3083989b1)

**Comment by christianlupus** [comment]
@gsantner unfortunately, there is more in the background to this. It is intentional to make everything available in the files app currently.

There have been considerations to change this but this might break existing scripts and tools. So, we have to be careful what we do in fact.

**Comment by gsantner** [comment]
> There have been considerations to change this but this might break existing scripts and tools. So, we have to be careful what we do in fact.

Of course! Though if you never get rid of a bad previous decision (in case it's viewed as such), or can't make any new decision because of such a restriction...you get decision paralyzed and end up doing nothing/worse after a while. Instead of making some breaking changes at some point to keep the project healthy and move on to touch other things.
